Course/Course/Data/CourseRepository.swift

```diff
@@ -255,6 +268,19 @@ class CourseRepositoryMock: CourseRepositoryProtocol {
         }
     }
     
+    func getCourseDeadlineInfo(courseID: String) async throws -> CourseDateBanner {
+        do {
+            let courseDates = try
+            await getCourseDates(courseID: courseID)
+            return CourseDateBanner(
+                datesBannerInfo: courseDates.datesBannerInfo,
+                hasEnded: courseDates.hasEnded
+            )
+        } catch {
+            throw error
+        }
```

Comment:
Do we need to catch an error just to throw it further?
Seems like we can simplify by removing the `do-catch` block.